internal/service/apigateway/integration.go

```diff
@@ -274,9 +274,7 @@ func resourceIntegrationRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("cache_namespace", integration.CacheNamespace)
 	d.Set("connection_id", integration.ConnectionId)
 	d.Set("connection_type", apigateway.ConnectionTypeInternet)
-	if integration.ConnectionType != nil {
-		d.Set("connection_type", integration.ConnectionType)
-	}
+	d.Set("connection_type", integration.ConnectionType)
```

Comment:
This is not right - the nil check is needed here.

Issues are arising due to this. See: https://github.com/hashicorp/terraform-provider-aws/issues/28997

Reply:
Fix in https://github.com/hashicorp/terraform-provider-aws/pull/29016.